pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -57,11 +57,10 @@ func (e *endpointsController) registerEndpointsHandler() {
 				})
 			},
 			UpdateFunc: func(old, cur interface{}) {
-				// Avoid pushes if only resource version changed (kube-scheduller, cluster-autoscaller, etc)
 				oldE := old.(*v1.Endpoints)
 				curE := cur.(*v1.Endpoints)
 
-				if !compareEndpoints(oldE, curE) {
+				if curE.ResourceVersion != oldE.ResourceVersion || !compareEndpoints(oldE, curE) {
```

Comment:
nit: can we move this condition also to the top of `compareEndpoints` before subset len check and add a test case for it?

Reply:
no, they are different stuff.

Reply:
is there ever any time the resource version is the same but the endpoints are different?

Is there ever a time when compareEndpoints gives a different response than the resource version check?

Why is update func even called if resource version changes?

Reply:
The version will not change when resynced

> is there ever any time the resource version is the same but the endpoints are different?

No

>Is there ever a time when compareEndpoints gives a different response than the resource version check?

The compareEndpoints is a little costly, and it is definitely a subset of version compare

>Why is update func even called if resource version changes?

I think you mean resource version not change, it is the k8s mechanism, to make sure client do not miss an object update as k8s is eventual consistent, and event may get lost  for many resons.